src/outputs/icecast2.ml

```diff
@@ -305,18 +308,21 @@ class output ~kind p =
 
   let source = Lang.assoc "" 2 p in
 
+  let icy_id =
+    Lang.to_int (List.assoc "icy_id" p)
+  in
+
   let mount = s "mount" in
   let name = s "name" in
   let name =
-    if name = no_name then
-      if mount = no_mount then
+    match protocol, name, mount with
+      | Cry.Http _, name, mount when name = no_name && mount = no_mount ->
         raise (Lang.Invalid_value
                  (List.assoc "mount" p,
-                  "Either name or mount must be defined"))
-      else
-        mount
-    else
-      name
+                  "Either name or mount must be defined for icecast sources."))
+      | Cry.Icy, name, _ when name = no_name -> Printf.sprintf "sc#%i" icy_id
```

Comment:
This doesn't seem to be backwards compatible: does it make sense if the protocol is icy but not v2?